File: Train-reservation-system-backend/src/main/java/com/Trainreservation/service/BookingService.java
package com.Trainreservation.service;

import java.util.List;

import com.Trainreservation.dto.TrainDetailsDto;
import com.Trainreservation.dto.TrainsDto;
import com.Trainreservation.dto.PassengerDto;
import com.Trainreservation.dto.SeatDto;
import com.Trainreservation.model.TrainDetails;
import com.Trainreservation.model.Seat;

public interface BookingService {

	List<TrainsDto> getAllTrainsAndSeatsDetails();

	List<TrainDetailsDto> getTrainByArrivalAndDeparture(String departure, String arrival, String departureDate);

	List<SeatDto> getAllSeatsFromATrain(String TrainNumber);

	PassengerDto bookTrainForAPassenger(TrainDetails TrainDetails, int userId);
	
	String bookSeatForAPassenger(Seat seat, String TrainNumber, int userId);

}

File: src/services/BookingService.js
import axios from "axios";

const BOOKING_API_BASE_URL = "http://localhost:8080/booking";

class BookingService {
  getTrainByArrivalAndDeparture(departure, arrival, departureDate) {
    return axios.get(
      BOOKING_API_BASE_URL +
        "/getTrainsOnArrivalAndDeparture/" +
        departure +
        "/" +
        arrival +
        "/" +
        departureDate
    );
  }

  getAllSeatsFromATrain(TrainNumber) {
    return axios.get(BOOKING_API_BASE_URL + "/getAllSeats/" + TrainNumber);
  }

  bookTrainForAPassenger(userId, TrainDetails) {
    return axios.post(
      BOOKING_API_BASE_URL + "/bookTrains/" + userId,
      TrainDetails
    );
  }

  bookSeatForAPassenger(TrainNumber, userId, seat) {
    return axios.post(
      BOOKING_API_BASE_URL + "/bookSeats/" + TrainNumber + "/" + userId,
      seat
    );
  }
}

export default new BookingService();
